game/svg.cc

```diff
@@ -34,19 +35,34 @@ void loadSVG(Bitmap& bitmap, fs::path const& filename) {
 	bitmap.resize(svgDimension.width*factor, svgDimension.height*factor);
 	bitmap.fmt = pix::INT_ARGB;
 	bitmap.linearPremul = true;
+    
+    RsvgPositionData pos;
+    rsvg_handle_get_position_sub (svgHandle.get(), &pos, nullptr);
 	// Raster with Cairo
 	std::shared_ptr<cairo_surface_t> surface(
-	  cairo_image_surface_create_for_data(&bitmap.buf[0], CAIRO_FORMAT_ARGB32, bitmap.width, bitmap.height, bitmap.width * 4),
+	  cairo_image_surface_create(CAIRO_FORMAT_ARGB32, bitmap.width, bitmap.height),
 	  cairo_surface_destroy);
 	std::shared_ptr<cairo_t> dc(cairo_create(surface.get()), cairo_destroy);
 	cairo_scale(dc.get(), factor, factor);
-	rsvg_handle_render_cairo(svgHandle.get(), dc.get());
+    cairo_translate (dc.get(), -pos.x, -pos.y);
+    rsvg_handle_render_cairo_sub(svgHandle.get(), dc.get(), nullptr);
+	// Copy rendered image to our buffer.
+	if (bitmap.ptr) throw std::logic_error("Bitmap shouldn't point to a foreign object.");
+	bitmap.copyFromCairo(surface.get());
+
 	// Change byte order from BGRA to RGBA
-	for (uint32_t *ptr = reinterpret_cast<uint32_t*>(&*bitmap.buf.begin()), *end = ptr + bitmap.buf.size() / 4; ptr < end; ++ptr) {
-		uint8_t* pixel = reinterpret_cast<uint8_t*>(ptr);
-		uint8_t r = pixel[2], g = pixel[1], b = pixel[0], a = pixel[3];
-		pixel[0] = r; pixel[1] = g; pixel[2] = b; pixel[3] = a;
+
+	for (uint8_t *ptr = &*bitmap.buf.begin(), *end = ptr + bitmap.buf.size(); ptr < end; ptr += 4) {
```

Comment:
```suggestion
	for (uint8_t *ptr = bitmap.buf.data(), *end = ptr + bitmap.buf.size(); ptr < end; ptr += 4) {
```

Reply:
Re: moving to to dedicated function... Maybe, I don't know how the others feel about it?
Re: your proposed change, is this a change needed for MSVC? I assume you tested it and it works properly. Yes?

Reply:
@Lord-Kamina To be honest, no, I still don't tested it yet, but vector::data is c++ 11 feature, should work almost everything: https://ru.cppreference.com/w/cpp/container/vector/data 
In theory data() method was designed specially to replace dereferencing of first element, and even replace undefined behavior when it is not present - the data will just return nullptr;

PS: let me try to build it with MXE :)

Reply:
@Lord-Kamina yes, it compiles and seems to work under MXE (after merging with master). 
So then I suggest to use vector.data() where it possible, especially if we plans to use new standards (seriously, today is already 2020, and we use 9 years-old standard :))

Reply:
I think technically we'd decided to target c++14 a while ago but still, point taken.

Reply:
Could very well go to C++17 already. Does the CI pipeline support that yet?

Reply:
I'm not certain. I think maybe semaphore 1 didn't ansd emaphore 2 did. The reason we were still using semaphore 1 is that pull requests didn't work properly on semaphore 2 but that has probably been fixed now.